File: server-scala-play-server/app/model/ThirdPartyNetworkResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for ThirdPartyNetworkResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class ThirdPartyNetworkResponse(
  updated: Option[Long],
  created: Option[Long],
  active: Option[Boolean],
  name: Option[String],
  description: Option[String],
  networkUID: Option[String],
  network: Option[ThirdPartyNetworkResponse.Network.Value],
  introspectionMethod: Option[ThirdPartyNetworkResponse.IntrospectionMethod.Value],
  enableIntrospection: Option[Boolean],
  introspectionURL: Option[String],
  introspectionParams: Option[String],
  requiredRootField: Option[String]
)

object ThirdPartyNetworkResponse {
  implicit lazy val thirdPartyNetworkResponseJsonFormat: Format[ThirdPartyNetworkResponse] = Json.format[ThirdPartyNetworkResponse]

  // noinspection TypeAnnotation
  object Network extends Enumeration {
    val FACEBOOK = Value("FACEBOOK")
    val TWITTER = Value("TWITTER")
    val CLEAR = Value("CLEAR")
    val USERNAME = Value("USERNAME")
    val PHONE = Value("PHONE")
    val UNKNOWN = Value("UNKNOWN")
    val CUSTOM = Value("CUSTOM")

    type Network = Value
    implicit lazy val NetworkJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }

  // noinspection TypeAnnotation
  object IntrospectionMethod extends Enumeration {
    val GET = Value("GET")
    val HEAD = Value("HEAD")
    val POST = Value("POST")
    val PUT = Value("PUT")
    val PATCH = Value("PATCH")
    val DELETE = Value("DELETE")
    val OPTIONS = Value("OPTIONS")
    val TRACE = Value("TRACE")

    type IntrospectionMethod = Value
    implicit lazy val IntrospectionMethodJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/ThirdPartyNetworkResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class ThirdPartyNetworkResponse 
{
    public long Updated { get; set; }
    public long Created { get; set; }
    public bool Active { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string NetworkUID { get; set; }
    
    /// <summary>
    /// Gets or Sets Network
    /// </summary>
    public enum NetworkEnum
    {
        
        /// <summary>
        /// Enum FACEBOOKEnum for FACEBOOK
        /// </summary>
        FACEBOOKEnum = 1,
        
        /// <summary>
        /// Enum TWITTEREnum for TWITTER
        /// </summary>
        TWITTEREnum = 2,
        
        /// <summary>
        /// Enum CLEAREnum for CLEAR
        /// </summary>
        CLEAREnum = 3,
        
        /// <summary>
        /// Enum USERNAMEEnum for USERNAME
        /// </summary>
        USERNAMEEnum = 4,
        
        /// <summary>
        /// Enum PHONEEnum for PHONE
        /// </summary>
        PHONEEnum = 5,
        
        /// <summary>
        /// Enum UNKNOWNEnum for UNKNOWN
        /// </summary>
        UNKNOWNEnum = 6,
        
        /// <summary>
        /// Enum CUSTOMEnum for CUSTOM
        /// </summary>
        CUSTOMEnum = 7
    }

    public NetworkEnum Network { get; set; }
    
    /// <summary>
    /// Gets or Sets IntrospectionMethod
    /// </summary>
    public enum IntrospectionMethodEnum
    {
        
        /// <summary>
        /// Enum GETEnum for GET
        /// </summary>
        GETEnum = 1,
        
        /// <summary>
        /// Enum HEADEnum for HEAD
        /// </summary>
        HEADEnum = 2,
        
        /// <summary>
        /// Enum POSTEnum for POST
        /// </summary>
        POSTEnum = 3,
        
        /// <summary>
        /// Enum PUTEnum for PUT
        /// </summary>
        PUTEnum = 4,
        
        /// <summary>
        /// Enum PATCHEnum for PATCH
        /// </summary>
        PATCHEnum = 5,
        
        /// <summary>
        /// Enum DELETEEnum for DELETE
        /// </summary>
        DELETEEnum = 6,
        
        /// <summary>
        /// Enum OPTIONSEnum for OPTIONS
        /// </summary>
        OPTIONSEnum = 7,
        
        /// <summary>
        /// Enum TRACEEnum for TRACE
        /// </summary>
        TRACEEnum = 8
    }

    public IntrospectionMethodEnum IntrospectionMethod { get; set; }
    public bool EnableIntrospection { get; set; }
    public string IntrospectionURL { get; set; }
    public string IntrospectionParams { get; set; }
    public string RequiredRootField { get; set; }
}


